ray_lightning/ray_ddp_sharded.py

```diff
@@ -0,0 +1,66 @@
+from typing import Optional
+
+import torch
+from torch.optim import Optimizer
+
+from pytorch_lightning import LightningModule
+from pytorch_lightning.utilities import _FAIRSCALE_AVAILABLE, rank_zero_only
+
+from ray_lightning import RayPlugin
+
+if _FAIRSCALE_AVAILABLE:
+    from fairscale.nn.data_parallel.sharded_ddp import ShardedDataParallel
+    from fairscale.optim import OSS
+
+    from pytorch_lightning.overrides.fairscale import \
+        LightningShardedDataParallel, unwrap_lightning_module_sharded
+
+
+class RayShardedPlugin(RayPlugin):
```

Comment:
would it be possible to do something like:

```
RayPlugin(mode="sharded")
```
?

Reply:
It is possible, though I would prefer to keep it as is. PTL has a separate `DDPPlugin` and `DDPShardedPlugin` so I think using the same dichotomy for these Ray plugins would be more intuitive for users.